### PR TITLE
Feature calendar

### PR DESCRIPTION
**Pull request by mathilde-pellerin**
- Update checkbox style
- Add popup directive and component
- Add dropdown components
- Add calendar and date picker components
- Add utility methods for popup positioning and markForCheck/detectChange methods

### CODE REVIEW

File: projects/pastanaga/src/lib/calendar/calendar.component.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        <div class="pa-calendar-navigation-buttons">
+            <pa-button icon="left-key" paTooltip="calendar.previous"
+                       (click)="goToPrevious()"></pa-button>
+            <pa-button icon="right-key" paTooltip="calendar.next"
+                       (click)="goToNext()"></pa-button>
+        </div>
+    </header>
+    <div class="pa-calendar-container">
+        <div class="pa-calendar-dates pa-week-days">
+            <ng-container *ngIf="view === 'day'">
```
[the remark this answers is not in the document]
Comment (mathilde-pellerin) [reply projects/pastanaga/src/lib/calendar/calendar.component.html]: I made a type `CalendarView` which is in `src/develop/pastanaga-angular/projects/pastanaga/src/lib/calendar/calendar.model.ts`, I just didn't imported it in a component property but I can if you think that would make it easier to understand

File: projects/pastanaga/src/lib/calendar/calendar.component.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                <div class="pa-calendar-date pa-calendar-day">S</div>
+                <div class="pa-calendar-date pa-calendar-day">M</div>
+                <div class="pa-calendar-date pa-calendar-day">T</div>
+                <div class="pa-calendar-date pa-calendar-day">W</div>
+                <div class="pa-calendar-date pa-calendar-day">T</div>
+                <div class="pa-calendar-date pa-calendar-day">F</div>
+                <div class="pa-calendar-date pa-calendar-day">S</div>
```
[the remark this answers is not in the document]
Comment (mathilde-pellerin) [reply projects/pastanaga/src/lib/calendar/calendar.component.html]: there would be more than a translation change to support different locales, so I chose to not do it for now. we will do that later if/when required

File: projects/pastanaga/src/lib/dropdown/dropdown-checkbox.component.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    @Output() onOpen: EventEmitter<void> = new EventEmitter();
+    @Output() onClose: EventEmitter<void> = new EventEmitter();
+
+    @ViewChild('menuRef', { read: PopupDirective, static: false }) menuRef?: PopupDirective;
+
+    labels: {[value: string]: string} = {};
+
+    constructor(private changeDetector: ChangeDetectorRef) {}
+
+    getlabels() {
```
Comment (barcafa) [inline projects/pastanaga/src/lib/dropdown/dropdown-checkbox.component.ts]: getLabels

File: projects/pastanaga/src/lib/dropdown/dropdown-item.component.html [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    </a>
+    <pa-dropdown *ngIf="subLevelItems?.length > 0" [id]="id + '-sublevels'">
+        <pa-dropdown-item *ngFor="let item of subLevelItems"
+                          [id]="item.id"
+                          [icon]="item.icon"
+                          [mode]="item.mode"
+                          [checkboxMode]="item.checkboxMode"
+                          [isSelected]="item.isSelected"
+                          (onSelection)="selectSubMenu(item)">
+            {{item.text | translate}}
```
Comment (barcafa) [inline projects/pastanaga/src/lib/dropdown/dropdown-item.component.html]: Is not dropdown-item already doing the translate?

File: projects/pastanaga/src/lib/popup/_popup.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    box-shadow: $shadow-pop;
+    border-radius: rythm(.25);
+    min-width: rythm(.5)*33;
+    max-width: rythm(1)*20;
+    max-height: rythm(1)*29;
+    overflow-x: hidden;
+    overflow-y: auto;
+
+    &-wrapper {
+        padding: 6px 0;
```
Comment (barcafa) [inline projects/pastanaga/src/lib/popup/_popup.scss]: rythm(.5)

File: src/app/app.component.ts [hunk 6]
```diff
@@ -256,4 +268,9 @@ export class AppComponent implements OnInit {
             menu.unfoldOnHover = !this.isLockedUnfolded;
         }
     }
+
+    hello() {
```
Comment (barcafa) [inline src/app/app.component.ts]: :)

File: projects/pastanaga/src/lib/dropdown/dropdown-item.component.html (hunk 4 above)
Comment (mathilde-pellerin) [reply projects/pastanaga/src/lib/dropdown/dropdown-item.component.html]: no, dropdown-item is displaying the label with `ng-content`:
```
<ng-template #labelContent><ng-content></ng-content></ng-template>
```

File: src/app/app.component.scss [hunk 7]
```diff
@@ -37,6 +48,9 @@
         border: 1px solid black;
         padding: 12px;
     }
+    h3, h4 {
+        margin-top: 24px;
```
Comment (barcafa) [inline src/app/app.component.scss]: rythm(2)

Comment (mathilde-pellerin) [reply src/app/app.component.scss]: that's just the demo project ;)

File: projects/pastanaga/src/lib/common/utils.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    let top = 0;
+    let left = 0;
+
+    while (!!tmp && tagName !== 'body') {
+        top += tmp.offsetTop;
+        if (ignoreSideMenu) {
+            left += tmp.offsetLeft;
+        } else {
+            const classes = tmp.getAttribute('class') || '';
+            if (!classes.includes('main-content') || !classes.includes('menu-open')) {
```
Comment (ebrehault) [inline projects/pastanaga/src/lib/common/utils.ts]: this is a custom case, we do not want that here

**Pull request merged**